test/Atmos/EDMF/helper_funcs/subdomain_thermo_states.jl

```diff
@@ -87,12 +87,19 @@ Recover the updraft thermodynamic states given:
     This method assumes that the temperature has been
     previously computed from a new thermodynamic state
     and stored in `aux`.
+
+!!! warn
+    While recover_thermo_state_up is an ideal long-term solution,
+    right now we are directly calling new_thermo_state_up to avoid
+    inconsistent aux states in kernels where the aux states are
+    out of sync with the boundary state.
 """
-recover_thermo_state_up(bl, state, aux) =
-    recover_thermo_state_up(bl, bl.moisture, state, aux)
+function recover_thermo_state_up(bl, state, aux, ts = new_thermo_state(bl, state, aux))
+    return new_thermo_state_up(bl, bl.moisture, state, aux, ts)
+end
```

Comment:
If `recover_thermo_state_up` calls for now `new_thermo_state_up`, why is this not used in `recover_thermo_state_all`? Same goes for the environment.

Reply:
If the reason is the warning message prior to this, I am not sure how the `aux` can be different if the same argument is passed in both cases.

Reply:
See issue #1648. Basically, `recover_thermo_state` is not a bullet-proof optimization. So, to be safe, this basically disables the optimization for now.